src/Integration.UnitTests/Connection/ConnectionWorkflowTests.cs

```diff
@@ -477,6 +468,11 @@ private void AssertCredentialsNotStored()
                 Times.Never());
         }
 
+        private static ConnectionWorkflow CreateTestSubject(IServiceProvider serviceProvider, IHost host, ICommand parentCommand)
+        {
+            return new ConnectionWorkflow(serviceProvider, host, parentCommand, new NoOpThreadHandler());
```

Comment:
I've hard-coded using a no-op thead handler here. There were no existing tests of the threading, and it would have been difficult to add any.

I manually debugged the product code and confirmed it is switching to/from the UI thread as expected.